### PR TITLE
REST API tests: Reorganize json-api unit tests

### PR DESCRIPTION
**Pull request by fgiannar**
Reorganize json-api unit tests

#### Changes proposed in this Pull Request:
* Not much, just re-organizing existing unit tests

#### Does this pull request change what data or activity we track or use?

no

#### Testing instructions:
n/a

#### Proposed changelog entry for your changes:

Changelog not needed as we are only reorganizing json-api related unit tests

### CODE REVIEW

File: tests/php/json-api/test-class.json-api-jetpack-endpoints.php [hunk 1]
```diff
@@ -1,38 +1,20 @@
...
 
 if ( ( ! defined( 'IS_WPCOM' ) || ! IS_WPCOM ) && defined( 'JETPACK__PLUGIN_DIR' ) && JETPACK__PLUGIN_DIR ) {
 	require_once JETPACK__PLUGIN_DIR . 'modules/module-extras.php';
 }
 
 require_jetpack_file( 'class.json-api-endpoints.php' );
 
+/**
+ * Tests for /sites/%s/categories/slug:%s
+ */
```
Comment (brbrr) [inline tests/php/json-api/test-class.json-api-jetpack-endpoints.php:16]: Do we need this comment here? It is the same as in the file header

**Review by brbrr — APPROVED** [review]
Left a minor comment. Otherwise, it looks good to me.

File: tests/php/json-api/test-class.json-api-jetpack-endpoints.php (hunk 1 above)
Comment (fgiannar) [reply tests/php/json-api/test-class.json-api-jetpack-endpoints.php:16]: @brbrr I was getting a `Missing doc comment for class` warning. Do you think it's best to remove it?

Comment (fgiannar) [reply tests/php/json-api/test-class.json-api-jetpack-endpoints.php:16]: Also I was trying to be consistent with `test-class.json-api-jetpack-site-endpoints.php`

Comment (brbrr) [reply tests/php/json-api/test-class.json-api-jetpack-endpoints.php:16]: Gocha, no worries then :)

**Review by brbrr — APPROVED** [review]

**Comment by fgiannar** [comment]
r214966-wpcom

**Pull request merged**